gaj23/bigbirds
# Gaj/feature/update sightings

#### Purpose: 
> Check all that apply.

### What does this PR do? 
> Clearly and concisely describe changes made. Include screenshots as needed.

This PR contains the largest part of this app: the ability for a new user to click on a bird in the top 30 list and see that bird added to their seen list. Additionally, if a bird exists on their seen list, then on the top 30 list, the bird should be crossed off to indicate to the user that they've seen that bird already. 

<img width="327" alt="seen list" src="https://user-images.githubusercontent.com/68332132/110385014-053d1800-8024-11eb-9291-e030214aa00e.png">
<img width="327" alt="top list cross off" src="https://user-images.githubusercontent.com/68332132/110385016-066e4500-8024-11eb-8932-b99690a98cb6.png">

This closes issue #12 

### Problems Encountered & Resulting Solutions  
> State any problems encountered & include screenshots as needed. 
> Describe your code changes in detail for reviewers.

A blocker was the management of data between components. The application of localStorage early on in `App` is what allowed to ensure that state is maintained and is accessible to children & sibling components. 

Another blocker was the application of a `checked` attribute. I realized most of my frustration was occurring because I had forgotten that I was rendering a static variable, rather than a function that reevaluates during rendering.

### Where should the reviewer start?
> What file(s) and line(s) are where changes are made?

A bulk of the work was done in `App` and `TopList`. To see problem solving revolving around updating and maintaining an accurate state, please look closely at the following functions: `checkList` and `possibleBirds`. 


### How should this be tested?
> What steps need to be taken to confirm functionality? 

Please clone down `git@github.com:gaj23/bigbirds.git` and after ensuring that npm is installed, type `npm start` in your CLI. Click through `Top 30` and note that birds already existing on the `Seen` list already are crossed off and if new birds are cross off, then they are also take off of the `Seen` list. Open the `Applications` tab in Chrome Dev tools to see that local storage is updating as expected. 


### Steps To Take
> Any questions and possible steps moving forward.

propTypes should be added to ensure dataType integrity. 
We should test for user flow with cypress. 
The highlighted bird should be randomized. 

#### Checklist:
